rasa/nlu/components.py

```diff
@@ -460,347 +466,75 @@ def name(self) -> Text:
 
-    @property
-    def unique_name(self) -> Text:
-        """Gets a unique name for the component in the pipeline.
-
-        The unique name can be used to distinguish components in
-        a pipeline, e.g. when the pipeline contains multiple
-        featurizers of the same type.
-        """
-        index = self.component_config.get(COMPONENT_INDEX)
-        return self.name if index is None else f"component_{index}_{self.name}"
-
-    @classmethod
-    def required_components(cls) -> List[Type["Component"]]:
-        """Specifies which components need to be present in the pipeline.
-
-        Which components are required by this component.
-        Listed components should appear before the component itself in the pipeline.
-
-        Returns:
-            The class names of the required components.
-        """
-        return []
-
-    # Defines the default configuration parameters of a component
-    # these values can be overwritten in the pipeline configuration
-    # of the model. The component should choose sensible defaults
-    # and should be able to create reasonable results with the defaults.
-    defaults = {}
-
-    # Defines what language(s) this component can handle.
-    # This attribute is designed for instance method: `can_handle_language`.
-    # Default value is None. if both `support_language_list` and
-    # `not_supported_language_list` are None, it means it can handle
-    # all languages. Also, only one of `support_language_list` and
-    # `not_supported_language_list` can be set to not None.
-    # This is an important feature for backwards compatibility of components.
-    supported_language_list = None
-
-    # Defines what language(s) this component can NOT handle.
-    # This attribute is designed for instance method: `can_handle_language`.
-    # Default value is None. if both `support_language_list` and
-    # `not_supported_language_list` are None, it means it can handle
-    # all languages. Also, only one of `support_language_list` and
-    # `not_supported_language_list` can be set to not None.
-    # This is an important feature for backwards compatibility of components.
-    not_supported_language_list = None
-
-    def __init__(self, component_config: Optional[Dict[Text, Any]] = None) -> None:
-
-        if not component_config:
-            component_config = {}
-
-        # makes sure the name of the configuration is part of the config
-        # this is important for e.g. persistence
-        component_config["name"] = self.name
-
-        self.component_config: Dict[
-            Text, Any
-        ] = rasa.utils.train_utils.override_defaults(self.defaults, component_config)
+    def __init__(
+        self,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> None:
+        """Initializes a new NLU Component."""
+        self.component_config: Dict[Text, Any] = config
```

Comment:
should this be protected?
```suggestion
        self._component_config: Dict[Text, Any] = config
```

Reply:
I think it's potentially useful to be able to access the config? And it's not something that solely inside the object

Reply:
It will definitely be useful if we can access that in the extra _validate steps..... 😁  But that doesn't mean we shouldn't protect the configs from modifications 🤔